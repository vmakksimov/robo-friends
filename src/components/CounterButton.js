import React, {PureComponent} from "react";



class CounterButton extends PureComponent {
    constructor(){
        super()

        this.state = {
            count : 0
        }

    }

    // shouldComponentUpdate(nextProps, nextState){
    //     if (this.state.count !== nextState.count){
    //         return true
    //     } else {
    //         return false
    //     }
      
    // }

    changeCounter = () => {
        this.setState(state => {
           return {count: state.count + 1}
        })
    }

    render(){
        return (
            <button id="counter" color={this.props.color} onClick={this.changeCounter} style={{color: this.props.color}}>Count : {this.state.count}</button>
        )
    }
}

export default CounterButton