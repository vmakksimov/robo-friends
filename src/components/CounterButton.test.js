import { shallow } from "enzyme";
import CounterButton from "./CounterButton";
import React from "react";


// it('counter button test', () => {
//     const mockColor = 'red'
//     expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
// })

// it('correctly icrements counter', () => {
//     const mockColor = 'red'
//     const wrapper = (shallow(<CounterButton color={mockColor}/>))
//     wrapper.find('[id="counter"]').simulate('click')
//     wrapper.find('[id="counter"]').simulate('click')
//     expect(wrapper.state()).toEqual({count: 2})
//     expect(wrapper.props().color).toEqual('red')
// })