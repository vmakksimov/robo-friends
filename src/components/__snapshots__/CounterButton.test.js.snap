// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter button test 1`] = `ShallowWrapper {}`;

exports[`testing shallow 1`] = `ShallowWrapper {}`;
